keyring/cli.py

```diff
@@ -90,6 +94,13 @@ def do_get(self):
             raise SystemExit(1)
         print(password)
 
+    def do_getcreds(self):
+        creds = get_credential(self.service, self.username)
+        if creds is None:
+            raise SystemExit(1)
+        print(f"username: {creds.username}")
+        print(f"password: {creds.password}")
```

Comment:
Should we just print them newline separated without the prefix?

Reply:
I am in favor of without prefix - I only included prefix because the only other command which outputs more than one item (`diagnose`) uses prefixes.

Reply:
What if the username or password contains a newline? Probably unlikely for both, but it's also maybe allowed?

FWIW, the `diagnose` interface is meant for human consumption so can be sloppy about syntax. I'm kinda leaning the same way for the rest of the keyring CLI API - meant primarily for human consumption. Still, no reason it can't provide a reasonably stable interface that a wrapper could rely upon.

I'm okay without the prefix as long as the help guides the user as to what they're getting from the output.

Reply:
Is it allowed for the username or password to include a newline?

The more complex option, I guess, is to output them base64 encoded.

Reply:
I looked at the tests and they do [test for difficult characters](https://github.com/jaraco/keyring/blob/35a3a384d333fc164949a6f85226209a922867bd/keyring/testing/backend.py#L35) on every keyring, so in theory they are supported. Still, I just wanted to flag it as a possibility.



> The more complex option, I guess, is to output them base64 encoded.

Oof. That wouldn't be user friendly. I'm thinking JSON would be better for safety, but that's not very friendly either. Maybe there should be an option to emit in an unambiguous form like JSON. (`--output json` (default 'plain'))?

Reply:
An output format enum seems reasonable.

Are the main users of the CLI humans? I presumed this was mostly used for programmatic interaction.

Reply:
Added `--output` argument

Help text for reference:
```
  --output {plain,json}
                        Output format for 'get' operation. Default is 'plain'
```

Example output:
```
 ➜  keyring --get-mode creds --output json get https://example.com
{"username": "hello", "password": "world"}
```

Reply:
> Are the main users of the CLI humans? I presumed this was mostly used for programmatic interaction.

I'm unsure. In my personal usage, I always use the CLI as a human and always use the library for programming, but that's because I'm always coding in Python. It's my impression that my experience has been the most common experience for a long time but that may be changing, especially with the introduction of keyring to homebrew and the use of it by tools like uv.

Reply:
> Added `--output` argument
> 
> Help text for reference:
> 
> ```
>   --output {plain,json}
>                         Output format for 'get' operation. Default is 'plain'
> ```
> 
> Example output:
> 
> ```
>  ➜  keyring --get-mode creds --output json get https://example.com
> {"username": "hello", "password": "world"}
>  ➜  keyring --get-mode creds get https://example.com
> hello
> world
> ```

I'm liking this a lot! Thanks. Is it possible to use `--mode` (since `get` is implied by the command)?

Reply:
Done

Reply:
(sorry I'm late to the party)

> > Are the main users of the CLI humans? I presumed this was mostly used for programmatic interaction.
>
> I'm unsure. In my personal usage, I always use the CLI as a human and always use the library for programming, but that's because I'm always coding in Python. It's my impression that my experience has been the most common experience for a long time but that may be changing, especially with the introduction of keyring to homebrew and the use of it by tools like uv.

If another datapoint is useful: pip has support for invoking `keyring` both [via python import](https://pip.pypa.io/en/stable/topics/authentication/#using-keyring-s-python-module) and [as a subprocess](https://pip.pypa.io/en/stable/topics/authentication/#using-keyring-as-a-command-line-application), and the author actually intended for ["most people to configure their system" with `subprocess`](https://github.com/pypa/pip/issues/12423).